<?xml version="1.0"?>
<launch>
  <!--
    Launch raspberry pi cameras using raspicam_node
    Republish the compressed image using:

    rosrun image_transport republish compressed in:=/<camera_name>_pod/color/image
    out:=/<camera_name>_pod/color/image_raw
  -->


  <arg name="private_topics"    default="false"/>
  <arg name="enable_raw"        default="false"/>
  <arg name="enable_imv"        default="false"/>
  <arg name="camera_id"         default="0"/>

  <arg name="camera_name"       default="camera2"/>
  <arg name="camera_frame_id"   default="$(arg camera_name)_color_optical_frame"/>
  <arg name="node_name"         default="raspicam_node"/>

  <arg name="width"     default="1296"/>
  <arg name="height"    default="972"/>
  <arg name="framerate" default="42"/>

  <group ns="$(arg camera_name)_pod">
    <node type="raspicam_node" pkg="raspicam_node" name="$(arg node_name)" output="screen" ns="color">
      <param name="private_topics" value="$(arg private_topics)"/>
      <param name="enable_raw" value="$(arg enable_raw)"/>
      <param name="enable_imv" value="$(arg enable_imv)"/>
      <param name="camera_id" value="$(arg camera_id)"/>

      <param name="camera_frame_id"   value="$(arg camera_frame_id)"/>
      <param name="camera_name"       value="$(arg camera_name)"/>
      <param name="camera_info_url"   value="package://raspicam_node/camera_info/$(arg camera_name)_$(arg width)x$(arg height).yaml"/>

      <param name="width"     value="$(arg width)"/>
      <param name="height"    value="$(arg height)"/>
      <param name="framerate" value="$(arg framerate)"/>
    </node>
  </group>

</launch>
